### PR TITLE
Java Collections & Attendance updating | Miguel A Valdes G

### PR DESCRIPTION
**Pull request by MiguelAValdesG**
Exercises of Java Collections from HackerRank done & updating of attendance list

### CODE REVIEW

File: MiguelAValdesG/Hackerrank/JavaArrayList.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import java.util.ArrayList;
```
Comment (luisod78) [inline MiguelAValdesG/Hackerrank/JavaArrayList.java:1]: no package name?

File: MiguelAValdesG/Hackerrank/JavaHashSet.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Scanner;
+
+public class JavaHashSet {
+
+  public static void main(String[] args) {
+    Scanner readLineTerminal = new Scanner(System.in);
+    HashSet<String> stringPairs = new HashSet<>();
```
Comment (luisod78) [inline MiguelAValdesG/Hackerrank/JavaHashSet.java:9]: Always using interface type when declaring a collection
https://www.codejava.net/java-core/collections/18-java-collections-and-generics-best-practices

File: MiguelAValdesG/Hackerrank/JavaHashSet.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import java.util.HashMap;
```
Comment (luisod78) [inline MiguelAValdesG/Hackerrank/JavaHashSet.java:1]: no package name?

File: MiguelAValdesG/Hackerrank/JavaHashTablesRansomNote.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import java.io.BufferedReader;
```
Comment (luisod78) [inline MiguelAValdesG/Hackerrank/JavaHashTablesRansomNote.java:1]: no package name?

File: MiguelAValdesG/Hackerrank/JavaHashTablesRansomNote.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+
+class Result {
+
+  public static boolean checkMagazine(List<String> magazine, List<String> note) {
+    HashMap<String, Integer> wordsInMagazine = new HashMap<>();
```
Comment (luisod78) [inline MiguelAValdesG/Hackerrank/JavaHashTablesRansomNote.java:13]: Always using interface type when declaring a collection
https://www.codejava.net/java-core/collections/18-java-collections-and-generics-best-practices

File: MiguelAValdesG/Hackerrank/JavaHashTablesRansomNote.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    final Integer INIT_VALUE = 0;
+    final Integer ONE = 1;
+    final boolean CAN_BE_FORMED = true;
+    final boolean CAN_NOT_BE_FORMED = false;
+
+    for (String word : magazine) {
+      if (!wordsInMagazine.containsKey(word)) {
+        wordsInMagazine.put(word, INIT_VALUE);
+      }
+      Integer wordConcurrences = wordsInMagazine.get(word);
```
Comment (luisod78) [inline MiguelAValdesG/Hackerrank/JavaHashTablesRansomNote.java:24]: wordConcurrences??? you mean occurrences right???

The more meaningful your code, the more readable

File: MiguelAValdesG/Hackerrank/JavaList.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+public class JavaList {
+
+  public static void main(String[] args) {
+    Scanner readLineTerminal = new Scanner(System.in);
+    int numberOfElementsInList = readLineTerminal.nextInt();
+    final String ERROR = "ERROR!";
+    final String SPACE = " ";
+    StringBuilder response = new StringBuilder();
+    LinkedList<Integer> list = new LinkedList<>();
```
Comment (luisod78) [inline MiguelAValdesG/Hackerrank/JavaList.java:13]: Always using interface type when declaring a collection
https://www.codejava.net/java-core/collections/18-java-collections-and-generics-best-practices

File: MiguelAValdesG/Hackerrank/JavaMap.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import java.util.HashMap;
+import java.util.Scanner;
+
+public class JavaMap {
+
+  public static void main(String[] args) {
+    Scanner readLineTerminal = new Scanner(System.in);
+    HashMap<String, Integer> phoneBook = new HashMap<>();
```
Comment (luisod78) [inline MiguelAValdesG/Hackerrank/JavaMap.java:8]: Always using interface type when declaring a collection
https://www.codejava.net/java-core/collections/18-java-collections-and-generics-best-practices

**Review by luisod78 — CHANGES_REQUESTED** [review]

File: MiguelAValdesG/HackerRank/fundamentals/JavaInheritanceII.java [hunk 9]
```diff
@@ -12,13 +14,13 @@ public int throwAdd(int firstNumber, int secondNumber) {
 
 class Solution{
   public static void main(String []args){
-    // Create a new Adder object
+    // Create a new fundamentals.Adder object
```
Comment (luisod78) [inline MiguelAValdesG/HackerRank/fundamentals/JavaInheritanceII.java:17]: please remove these comments

File: MiguelAValdesG/HackerRank/collections/JavaHashTablesRansomNote.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+
+class Result {
```
Comment (luisod78) [inline MiguelAValdesG/HackerRank/collections/JavaHashTablesRansomNote.java:13]: Each class must be in its own file

**Review by luisod78 — CHANGES_REQUESTED** [review]

**Review by luisod78 — APPROVED** [review]

**Review by edbermeo — APPROVED** [review]
LGTM!

**Pull request merged**